### PR TITLE
feat (api): add manual trigger

### PR DESCRIPTION
**Pull request by sguiheux**

### CODE REVIEW

File: engine/api/workflow/dao.go [hunk 1]
```diff
@@ -145,12 +152,92 @@ func Insert(db gorp.SqlExecutor, store cache.Store, w *sdk.Workflow, u *sdk.User
...
+	// Save node in pipeline node map
+	if _, ok := (*dict)[pipID]; !ok {
+		(*dict)[pipID] = []*sdk.WorkflowNode{}
+	}
+	(*dict)[pipID] = append((*dict)[pipID], n)
+
+	// Check max number for current pipeline
+	if n.Name != "" && strings.HasPrefix(n.Name, pip.Name+"_") {
+		pipNumber, errI := strconv.ParseInt(strings.Replace(n.Name, pip.Name+"_", "", 1), 10, 64)
+		if errI == nil {
```
Comment (yesnault) [inline engine/api/workflow/dao.go]: log.error if errI is not nil, no?

File: engine/api/workflow/dao.go [hunk 2]
```diff
@@ -145,12 +152,92 @@ func Insert(db gorp.SqlExecutor, store cache.Store, w *sdk.Workflow, u *sdk.User
...
+			currentMax, ok := (*mapMaxNumber)[pipID]
+			if !ok || currentMax < pipNumber {
+				(*mapMaxNumber)[pipID] = pipNumber
+			}
+		}
+	}
+
+	for k := range n.Triggers {
+		if err := saveNodeByPipeline(db, dict, mapMaxNumber, &n.Triggers[k].WorkflowDestNode); err != nil {
+			return err
```
Comment (yesnault) [inline engine/api/workflow/dao.go]: wrap?

Comment (sguiheux) [reply engine/api/workflow/dao.go]: no,

File: engine/api/workflow/dao.go (hunk 1 above)
Comment (sguiheux) [reply engine/api/workflow/dao.go]: no

**Pull request merged**